doc: Extended Matter configuring protocol section (#6748)
nrfconnect/sdk-nrf: doc/nrf/ug_matter_configuring_protocol.rst

See :doc:`matter:nrfconnect_examples_cli` in the Matter documentation for the list of available Matter shell commands.

.. _ug_matter_configuring_device_identification:

Matter device identification
============================

> Matter has many mandatory and optional ways to identify a specific device.

Reviewer: ```suggestion
Matter has many ways to identify a specific device, both mandatory and optional .
```

These can be used for various purposes, such as dividing devices into groups (by function, by vendor or by location), device commissioning or vendor-specific cases before the device was commissioned (for example, identifying factory software version or related features).

> Only some part of these features can be configured using Kconfig options and only those were listed below:

Reviewer: ```suggestion
Some of these can be configured using the Kconfig options listed below:
```

> * :kconfig:`CONFIG_CHIP_DEVICE_TYPE` - type of device that uses the Matter Device Type Identifier, for example Door Lock (0x000A) or Dimmable Light Bulb (0x0101).
> * :kconfig:`CONFIG_CHIP_COMMISSIONABLE_DEVICE_TYPE` - enables including optional device type subtype in the commissionable node discovery record, which allows filtering of the discovery results to find the nodes that match the device type.
> * :kconfig:`CONFIG_CHIP_ROTATING_DEVICE_ID` - enables rotating device identifier, an optional feature that provides an additional unique identifier for each device. This identifier is similar to the serial number, but it additionally changes at predefined times to protect against long-term tracking of the device.

Reviewer: ```suggestion
* :kconfig:`CONFIG_CHIP_DEVICE_TYPE` sets the type of the device using the Matter Device Type Identifier, for example Door Lock (0x000A) or Dimmable Light Bulb (0x0101).
* :kconfig:`CONFIG_CHIP_COMMISSIONABLE_DEVICE_TYPE` enables including an optional device type subtype in the commissionable node discovery record.
  This allows filtering of the discovery results to find the nodes that match the device type.
* :kconfig:`CONFIG_CHIP_ROTATING_DEVICE_ID` enables an optional rotating device identifier feature that provides an additional unique identifier for each device.
  This identifier is similar to the serial number, but it additionally changes at predefined times to protect against long-term tracking of the device.
```
Should the second bullet end with "[...] that match the device subtype"?

Author: I meant that by default those discovery records doesn't have information about device type. So if you have plenty of devices you don't really know what kind of device they are. In case you enable it it is possible to filter bunch of records to find only those that match desired device type (for example light bulb or temperature sensor)

Reviewer: Ah, then it can stay as it is.
